2_correlation_analysis/spell_vs_counts_experiment/1_correlation_analysis.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:core_acc] *
+#     language: python
+#     name: conda-env-core_acc-py
+# ---
+
+# # Correlation analysis
+#
+# This notebook creates the correlation matrix using the MR counts and a SPELL processed version.
+
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+import os
+import pandas as pd
+import plotnine as pn
+import seaborn as sns
+from sklearn import preprocessing
+import matplotlib.pyplot as plt
+import umap
+import random
+import numpy as np
+import scipy
+from scipy.spatial.distance import pdist, squareform
+from sklearn.decomposition import PCA
+from scripts import paths, utils
+
+# ## Set user parameters
+#
+# Here we set the number of PCs or singular vectors to use. We are starting with 300 since this is what [eADAGE](https://pubmed.ncbi.nlm.nih.gov/28711280/) used.
+
+# +
+# Params
+
+# Which subset of genes to consider: core, acc, all
+subset_genes = "all"
+
+# The number of accessory genes is 200 - 500
+# The number of core genes is ~ 5000
+# So the number of singular vectors is relative to the number of genes
+if subset_genes == "acc":
```

Comment:
How did you choose 50 and 100 here? Could you look at the variance explained by this many SVs and make sure they're similar for each dataset?

Reply:
This was based on a manual inspection of the heatmaps -- examining the size of the clusters and make sure the very dominant signal was removed. I can also look at the variance explained here and confirm that the results are similar to the numbers chosen

Reply:
Up to you, if you've already looked at cluster size and confirmed the results are what you expect then maybe there's no real need.

Reply:
I've added a comment to clarify how these were chosen. Its pretty quick to plot that so I think I'll just add it just to make sure that the numbers align with my observations